frontend/packages/dev-console/src/components/topology/GraphContextMenu.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import { PopupKebabMenu } from '@console/shared/src';
+import { KebabOption } from '@console/internal/components/utils';
+import { ActionProvider, GraphElementType } from './topology-types';
+
+export interface GraphContextMenuProps {
+  actionProvider: ActionProvider;
+}
+
+interface MenuState {
+  isOpen: boolean;
+  kebabOptions: KebabOption[];
+  eventX: number;
+  eventY: number;
+}
+
+export class GraphContextMenu extends React.Component<GraphContextMenuProps, MenuState> {
+  private menuContainer;
+
+  private modalContainer;
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: false,
+      kebabOptions: null,
+      eventX: 0,
+      eventY: 0,
+    };
+  }
+
+  componentDidMount() {
+    this.modalContainer = document.getElementById('modal-container');
+  }
+
+  public open = (type: GraphElementType, id: string, eventX: number, eventY: number) => {
+    const { actionProvider } = this.props;
+
+    const kebabOptions = actionProvider(type, id);
+    const isOpen = kebabOptions && kebabOptions.length > 0;
+
+    this.setState({ isOpen, kebabOptions, eventX, eventY });
+
+    return isOpen;
+  };
+
+  public onClose = () => {
+    this.setState({ isOpen: false });
+  };
+
+  renderContextMenu() {
+    const { kebabOptions, eventX, eventY } = this.state;
+
+    return (
+      <PopupKebabMenu
+        onClose={this.onClose}
+        container={this.menuContainer}
+        kebabOptions={kebabOptions}
+        eventX={eventX}
+        eventY={eventY}
+      />
+    );
+  }
+
+  render() {
+    const { isOpen } = this.state;
+
+    return (
+      <div className="odc-graph-context-menu" ref={(ref) => (this.menuContainer = ref)}>
```

Comment:
Since this is not a `PureComponent`, make the anonymous ref function an instance function on the class.